src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -703,6 +704,25 @@ public DatasetSummary mapRow(ResultSet rs, int rowNum) throws SQLException {
     }
   }
 
+  /**
+   * Update a dataset according to specified fields in the patch request. Any fields unspecified in
+   * the request will remain unaltered.
+   *
+   * @param id dataset UUID
+   * @param patchRequest updates to merge with existing dataset
+   * @return whether the dataset record was updated
+   */
+  @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+  public boolean patch(UUID id, DatasetPatchRequestModel patchRequest) {
+    String sql = "UPDATE dataset SET phs_id = COALESCE(:phs_id, phs_id) WHERE id = :id";
```

Comment:
I'm curious about this use of `COALESCE`. My understanding is that `COALESCE(x, y, z)` returns the first non-null object. So if `patchRequest.getPhsId()` is `null`, then this would be a no-op. If that's the desired behavior, could you get the same result by skipping the `update()` call when `patchRequest.getPhsId()` is `null`?

Reply:
Thanks for asking.  That's my understanding of `COALESCE` as well.  I like using it here because we'll soon expand these endpoints to allow other elements of the dataset / snapshot to be modified, this way we'll keep the existing values for any `patchRequest` elements left unspecified.

A concrete example, if/when we allow dataset descriptions to be modified:
```
UPDATE dataset SET
  phs_id = COALESCE(:phs_id, phs_id),
  description = COALESCE(:description, description)
WHERE id = :id"
```

A user could then overwrite the description without modifying the PHS ID.